enumerable.rb

```diff
@@ -0,0 +1,160 @@
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    index = 0
+    while index < size
+      yield(self[index])
+      index += 1
+    end
+    self
+  end
+
+  def my_each_with_index(given_index = nil)
+    return enum_for(:my_each_with_index) unless block_given?
+
+    index = 0
+    index = given_index unless given_index.nil?
+    my_each do |x|
+      yield(x, index)
+      index += 1
+    end
+    self
+  end
+
+  def my_select
+    select_array = []
+    return enum_for(:my_select) unless block_given?
+
+    my_each { |x| select_array << x if yield x }
+    select_array
+  end
+
+  def my_all?(pattern = nil)
+    my_each do |x|
+      if block_given?
+        return false unless yield x
+      elsif !pattern.nil?
+        return false if all_check(pattern, x) == false
+      else
+        return false unless x
+      end
+    end
+    true
+  end
+
+  def my_any?
+    result = false
+    my_each do |i|
+      result = true if yield i
+      break if result
+    end
+    result
+  end
```

Comment:
**`my_any?`**  identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- when no block or argument is given returns true if at least one of the collection is not false or nil
```ruby
[1, false, nil].any?  #should return true.
- when a class is passed as an argument returns true if at least one of the collection is a member of such class
```ruby
[1, 'class', nil].any?(Integer)  #should return true
```
- when a Regex is passed as an argument returns false if none of the collection matches the Regex
```ruby
%w[hello hi howdy].any?(/d/) #should return false
%w[hello11 hi howdy].any?(/d/) #should return true
```
- when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
```ruby
[1, 2, 3].any?(3)  #should return true
```
While playing with those failing cases came up with a solution as below. Had to change current my_any? with the following code and updated the any_check method(commented below). And it worked.
```ruby
  def my_any?(pattern = nil)
    my_each do |x|
      if block_given?
        return true if yield x
      elsif !pattern.nil?
        return true if any_check(pattern, x)
      elsif x
        return true
      end
    end
    false
  end
```